app/controllers/v0/sign_in_controller.rb

```diff
@@ -330,12 +330,23 @@ def create_login_code(state_payload, user_info, credential_level) # rubocop:disa
                               "client_id:#{state_payload.client_id}",
                               "ial:#{credential_level.current_ial}",
                               "acr:#{state_payload.acr}"])
-      params_hash = { code: user_code_map.login_code, type: user_code_map.type }
-      params_hash.merge!(state: user_code_map.client_state) if user_code_map.client_state.present?
+      handle_login_code_redirect(user_code_map, state_payload.client_id)
+    end
 
-      render body: SignIn::RedirectUrlGenerator.new(redirect_uri: user_code_map.client_config.redirect_uri,
-                                                    params_hash:).perform,
-             content_type: 'text/html'
+    def handle_login_code_redirect(user_code_map, client_id)
+      params_hash = { code: user_code_map.login_code,
+                      type: user_code_map.type,
+                      state: user_code_map.client_state }.compact
+
+      if cookie_authentication?(client_id)
+        render body: SignIn::RedirectUrlGenerator.new(redirect_uri: user_code_map.client_config.redirect_uri,
+                                                      params_hash:).perform,
+               content_type: 'text/html'
+      else
+        redirect_uri = URI.parse(user_code_map.client_config.redirect_uri)
+        redirect_uri.query = params_hash.to_query
+        redirect_to(redirect_uri.to_s)
+      end
```

Comment:
Is the Android client the only client configured as something other than cookie-based authentication? If not, this would introduce the change across a wider surface area.

Would it make sense to modify this to be more selective? One suggestion would be to target the change exclusively to the Android client.

```ruby
      if client_id.casecmp?('Android') # I'm sure there is probably a better way
        redirect_uri = URI.parse(user_code_map.client_config.redirect_uri)
        redirect_uri.query = params_hash.to_query
        redirect_to(redirect_uri.to_s)
      else
        render body: SignIn::RedirectUrlGenerator.new(redirect_uri: user_code_map.client_config.redirect_uri,
                                                      params_hash:).perform,
               content_type: 'text/html'
      end
```